include/llbuild/BuildSystem/BuildValue.h

```diff
@@ -52,15 +52,21 @@ class BuildValue {
 
     /// A value for a produced output whose command failed.
     FailedInput,
+      
+    /// A value for a produced output whose command was cancelled.
+    CancelledInput,
 
     /// A value produced by a successful command.
     SuccessfulCommand,
 
     /// A value produced by a failing command.
     FailedCommand,
-      
-    /// A value produced by a command which was skipped.
-    SkippedCommand,
+
+    /// A value produced by a command which was skipped because one of its dependencies failed.
+    FailedDependencyCommand,
```

Comment:
UnavailableCommand reads well, but I wonder if we can find something that's terse yet still completely explicit about what happened? PropagatedFailureCommand?